index.js

```diff
@@ -84,11 +85,21 @@ function showError(err, req, res, next) { // eslint-disable-line no-unused-vars
   // public consumption.
   if (err instanceof ErrorHTTP && err.status < 500) {
     for (const k in err) {
-      if (k === 'status') continue;
+      if (k === 'status' || k === 'ttl') continue;
       data[k] = err[k];
     }
   }
 
+  // Allow downstream APIs to provide custom cache-control values
+  // for their errors, while also ensure that we respect any custom
+  // cache-control values set at the ErrorHTTP level.
+  if (ttl && !err.ttl) {
```

Comment:
I understand it's their fault if they don't do it right, but should we syntax check the ttl value (here and in the constructor)?  This has very defined syntax (as opposed to status just being a number, and message being some arbitrary text)

Reply:
Yeah, I considered doing this at first, but ultimately set it aside because of how straight forward the library is. Happy to add some syntax checking if you feel strongly about this being more rigorous 👍

Reply:
I think some syntax checking would be good, as we would rather this fail in the user's unit testing (or fail early) if they screwed up the cache value, rather than sending out corrupt HTTP responses.

Reply:
Great point. I don't think we should do any throwing from within `ErrorHTTP` because this could cause users to accidentally pass a validation error to their express error handler and not realize it. 

However, throwing from within the `showError` custom error handler doesn't feel like an issue and could be very useful for anyone upgrading or implementing this version.

Reply:
Added some errors in https://github.com/mapbox/mapbox-error/pull/19/commits/5afb1da82ee5ddf0c333d869b909cb7914cb8f48. Let me know if this is inline with your thinking @corporategoth 👍